modules/uid2IdSystem_shared.js

```diff
@@ -36,49 +146,154 @@ export class Uid2ApiClient {
   }
   callRefreshApi(refreshDetails) {
     const url = this._baseUrl + '/v2/token/refresh';
-    const req = new XMLHttpRequest();
-    req.overrideMimeType('text/plain');
-    req.open('POST', url, true);
-    req.setRequestHeader('X-UID2-Client-Version', this._clientVersion);
     let resolvePromise;
     let rejectPromise;
     const promise = new Promise((resolve, reject) => {
       resolvePromise = resolve;
       rejectPromise = reject;
     });
-    req.onreadystatechange = () => {
-      if (req.readyState !== req.DONE) { return; }
-      try {
-        if (!refreshDetails.refresh_response_key || req.status !== 200) {
-          this._logInfo('Error status OR no response decryption key available, assuming unencrypted JSON');
-          const response = JSON.parse(req.responseText);
+    this._logInfo('Sending refresh request', refreshDetails);
+    ajax(url, {
```

Comment:
Would using [`fetch`](https://github.com/prebid/Prebid.js/blob/5fc475ed7423dcc295dafc6a552097b191f049d5/src/ajax.js#L138) be preferable to `ajax`?

Reply:
Lionell got pinged from Prebid  - apparently we must use their ajax wrapper for any calls we make

Reply:
Just clarifying I meant Prebid's `fetch` wrapper. Is that not allowed?

Reply:
yes, we can use Prebid's `fetch`, but I think ajax has the timeout and callback wrap, I think we should use `ajax` instead?
```
export function ajaxBuilder(timeout = 3000, {request, done} = {}) {
  const fetcher = fetcherFactory(timeout, {request, done});
  return function (url, callback, data, options = {}) {
    attachCallbacks(fetcher(toFetchRequest(url, data, options)), callback);
  };
}

export const ajax = ajaxBuilder();
export const fetch = fetcherFactory();

```